batcher/aligned-batcher/src/lib.rs

```diff
@@ -1232,25 +1250,63 @@ impl Batcher {
                 }
                 Ok(receipt)
             }
-            Err(BatcherSendError::TransactionReverted(err)) => {
-                // Since transaction was reverted, we don't want to retry with fallback.
-                warn!("Transaction reverted {:?}", err);
-
-                Err(BatcherError::TransactionSendError)
+            Err(RetryError::Permanent(BatcherError::ReceiptNotFoundError)) => {
+                self.metrics.canceled_batches.inc();
+                self.cancel_create_new_task_tx(fee_params.gas_price).await;
+                Err(BatcherError::ReceiptNotFoundError)
             }
-            Err(_) => {
-                let receipt = try_create_new_task(
-                    batch_merkle_root,
-                    batch_data_pointer,
-                    proof_submitters,
-                    fee_params,
-                    &self.payment_service_fallback,
+            Err(_) => Err(BatcherError::TransactionSendError),
+        }
+    }
+
+    // Sends a transaction to Ethereum with the same nonce as the previous one to override it.
+    // Uses exponential backoff with the maximum number of retries because this transaction must be included.
+    // Bumps the gas price used and waits for 3 blocks on each attepmt to get the Receipt of the cancel transaction.
```

Comment:
```suggestion
    /// Sends a transaction to Ethereum with the same nonce as the previous one to override it.
    /// In case of a recoverable error, it will retry with an exponential backoff forever
    /// A tx not included in 3 blocks will be considered an error, and will trigger a bump of the fee, with the rules on ```get_bumped_gas_price```
    /// This will do 5 bumps every 3 blocks, and then the exponential backoff will dominate, doiong bumps at 8,13,24,45,89 and so on
```